perses/rjmc/geometry.py

```diff
@@ -96,46 +97,49 @@ def __init__(self, metadata=None, use_sterics=False, verbose=True):
         self.verbose = verbose
         self.use_sterics = use_sterics
 
-    def propose(self, top_proposal, current_positions, beta):
+    def propose(self, top_proposal, current_sampler_state, beta):
         """
         Make a geometry proposal for the appropriate atoms.
 
         Arguments
         ----------
         top_proposal : TopologyProposal object
             Object containing the relevant results of a topology proposal
+        current_sampler_state : openmmtools.states.SamplerState
+            The current positions and box vectors of the proposal
         beta : float
             The inverse temperature
 
         Returns
         -------
-        new_positions : [n, 3] ndarray
+        new_sampler_state : openmmtools.states.SamplerState
             The new positions of the system
         logp_proposal : float
             The log probability of the forward-only proposal
         """
+        current_positions = current_sampler_state.positions
         current_positions = current_positions.in_units_of(units.nanometers)
         if not top_proposal.unique_new_atoms:
             structure = parmed.openmm.load_topology(top_proposal.old_topology, top_proposal.old_system)
             atoms_with_positions = [structure.atoms[atom_idx] for atom_idx in top_proposal.new_to_old_atom_map.keys()]
             new_positions = self._copy_positions(atoms_with_positions, top_proposal, current_positions)
             return new_positions, 0.0
```

Comment:
This line should return a `SamplerState`, not positions.

Should `_copy_positions()` also be converted to use `SamplerState`? `_copy_positions()` seems a bit kludgey. Is there an easier way to replace its functionality with a method in`SamplerState` that either exists or we could add?

Reply:
Good point. Maybe there's an easier way to do this directly with `SamplerState`s.